backend/lib-shared-site-api/src/cache/cache.rs

```diff
@@ -33,11 +36,35 @@ pub type SiteMetadataCache = Cache<String, SiteMetadata>; // key: site_id, value
 #[derive(Clone)]
 pub struct AppCache {
     pub cache: SiteUsageCache,
-    domain_cache: SiteDomainCache,
-    metadata_cache: SiteMetadataCache,
+    pub domain_cache: SiteDomainCache,
+    pub metadata_cache: SiteMetadataCache,
     exec_env: ExecEnv,
 }
 
+fn site_to_value(site: &SiteEntity) -> Result<GetCurrentSiteResponse, serde_json::Error> {
+    let unwrapped_pages: String = serde_json::from_str(&site.pages)?;
+    let mut pages: Map<String, Value> = serde_json::from_str(&unwrapped_pages)?;
+
+    let mut filtered_json = Map::new();
+    for (key, value) in pages.iter_mut() {
+        if let Some(public) = value.get("public").and_then(Value::as_bool) {
+            if public {
+                filtered_json.insert(key.clone(), value.clone());
+            }
+        }
+    }
+    let pages_json = serde_json::to_string(&filtered_json)?;
+    let filtered_pages = serde_json::to_string(&pages_json)?;
+    Ok(GetCurrentSiteResponse {
```

Comment:
I changed the cache to store `GetCurrentSiteResponse`, that way we don't need to do any processing in the endpoint.